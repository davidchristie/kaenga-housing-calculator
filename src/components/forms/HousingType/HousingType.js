import React, { Component, PropTypes } from 'react'
import { FormGroup, Panel } from 'react-bootstrap'
import { Field } from 'redux-form'

import Select from '../../controls/Select'
import Anchor from '../../layout/Anchor'
import Tip from '../../misc/Tip'

class HousingType extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel header='Housing Type'>
          <Anchor id='housing-type' />
          <Tip name='housing-types'>
            In most locations, terraced housing and low rise apartments offer the best value for money
          </Tip>
          <FormGroup>
            <Field component={Select} name='type'>
              <option value='house'>
                Standalone house
              </option>
              <option value='terraced'>
                Terraced house (3-10 homes)
              </option>
              <option value='lowRise'>
                Low rise apartments (12-25 homes)
              </option>
              <option value='highRise'>
                High rise apartments (30+ homes)
              </option>
            </Field>
          </FormGroup>
        </Panel>
      </form>
    )
  }
}

HousingType.propTypes = {
  handleSubmit: PropTypes.func
}

export default HousingType
