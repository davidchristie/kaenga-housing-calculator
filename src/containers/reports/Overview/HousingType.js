import React, { Component } from 'react'
import { connect } from 'react-redux'

import housingType from '../../../selectors/housing-type'
import Box from './Box'

class HousingType extends Component {
  render () {
    const { values } = this.props
    const terraced = values.terracedHousing
    const apartment = Math.max(
      values.highRise,
      values.midScaleApartment,
      values.largeScaleApartment
    )
    let type = null
    if (terraced === apartment) {
      type = 'Terraced housing or apartment'
    } else if (terraced > apartment) {
      type = 'Terraced housing'
    } else {
      type = 'Apartment'
    }
    return (
      <Box>
        <h4>Housing Type</h4>
        <strong>{type}</strong>
      </Box>
    )
  }
}

export default connect(
  state => {
    return {
      values: housingType(state)
    }
  }
)(HousingType)
