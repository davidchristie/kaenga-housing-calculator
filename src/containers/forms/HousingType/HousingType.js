import { reduxForm } from 'redux-form'

import HousingType from '../../../components/forms/HousingType'

export default reduxForm({
  destroyOnUnmount: false,
  form: 'housingType',
  initialValues: {
    type: ''
  }
})(HousingType)
