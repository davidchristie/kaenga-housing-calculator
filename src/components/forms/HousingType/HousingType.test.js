import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import HousingType from './HousingType'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<HousingType />)).toMatchSnapshot()
})
