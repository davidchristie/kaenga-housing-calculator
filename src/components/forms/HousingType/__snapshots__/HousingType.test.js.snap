exports[`test matches snapshot 1`] = `
<form
  onSubmit={undefined}>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <Anchor
      id="housing-type" />
    <h3>
      Type of Home
       
      <Tip
        name="housing-types">
        We only build terraced housing and low rise apartments because they offer best value for money
      </Tip>
    </h3>
    <FormGroup
      bsClass="form-group">
      <Field
        component={[Function]}
        name="type">
        <option
          disabled={true}
          hidden={true}
          value="">
          Please select...
        </option>
        <option
          value="terraced">
          Terraced house (3-10 homes)
        </option>
        <option
          value="lowRise">
          Low rise apartments (12-25 homes)
        </option>
      </Field>
    </FormGroup>
  </Panel>
</form>
`;
